### PR TITLE
Improve EGun device

### PR DESCRIPTION
**Pull request by anacso17**
Add first version of methods to set EGun HVPS voltage and FilaPS Current (partially based on old scripts used in commissioning)

### CODE REVIEW

File: siriuspy/siriuspy/devices/egun.py [hunk 1]
```diff
@@ -132,6 +135,14 @@ def is_on(self):
         """."""
         return self['swstatus'] == self.PWRSTATE.On
 
+    def wait_current(self, value, tol=0.20, timeout=DEF_TIMEOUT):
+        """Wait current to reach value with tolerance 'tol'."""
+        _t0 = _time.time()
+        while _time.time() - _t0 < timeout:
+            if abs(self.current - value) < tol:
+                return True
```
Comment (xresende) [inline siriuspy/siriuspy/devices/egun.py:143]: @anacso17 , don't you think we should insert a `time.sleep` in the loop in order to ease on the cpu ?

File: siriuspy/siriuspy/devices/egun.py [hunk 2]
```diff
@@ -204,6 +215,14 @@ def is_on(self):
         ison &= self['swstatus'] == self.PWRSTATE.On
         return ison
 
+    def wait_voltage(self, value, tol=1, timeout=DEF_TIMEOUT):
+        """Wait voltage to reach value with tolerance 'tol'."""
+        _t0 = _time.time()
+        while _time.time() - _t0 < timeout:
```
Comment (xresende) [inline siriuspy/siriuspy/devices/egun.py:221]: same here

**Review by xresende — APPROVED** [review]

File: siriuspy/siriuspy/devices/egun.py (hunk 1 above)
Comment (anacso17) [reply siriuspy/siriuspy/devices/egun.py:143]: thanks!!

**Review by murilobalves — APPROVED** [review]

File: siriuspy/siriuspy/devices/egun.py [hunk 3]
```diff
@@ -406,8 +427,15 @@ class EGun(_Devices):
     BIAS_TOLERANCE = 1.0  # [V]
     HV_OPVALUE = 90.0  # [V]
     HV_TOLERANCE = 1.0  # [V]
+    HV_LEAKCURR_OPVALUE = 0.015  # [mA]
+    HV_MAXVALUE = 90.0  # [V]
```
Comment (murilobalves) [inline siriuspy/siriuspy/devices/egun.py:431]: HV unit: [V] -> [kV]

**Review by murilobalves — APPROVED** [review]

**Review by fernandohds564 — APPROVED** [review]

File: siriuspy/siriuspy/devices/egun.py (hunk 3 above)
Comment (anacso17) [reply siriuspy/siriuspy/devices/egun.py]: Thanks, @murilobalves

**Review by xresende — APPROVED** [review]

**Pull request merged**